[v3.0.12] Bugfixes, Code Cleanup & Deprecation

KuryKat opened the pull request:
fixes [#46](https://github.com/hypherionmc/sdlink/issues/46) + Code Cleanup + Deprecation of Text Commands

KuryKat committed:
[BUG] Text LinkCommand using incorrect args index

fixes https://github.com/hypherionmc/sdlink/issues/46

KuryKat committed:
[BUG] Text LinkCommand don't set linkedRole

Also removed some unused code on Slash LinkCommand

KuryKat committed:
[FEAT] Code Cleanup LinkCommands

(Both Slash and Text commands)

Code cleanup to remove code that was meant to be on the linkAccount method.

KuryKat committed:
[CHORE] Bump Library Version

KuryKat committed:
[BUG] Fix Missing arguments

Ooopsies eheh XD

KuryKat committed:
[BUG] botStatusType "STREAMING" not working

The bot wouldn't show the status correctly because it was missing the required URL to display as "streaming".

Also updated a few comments for better understanding of its use.

KuryKat committed:
[BUG] Link Account not working

Fix several bugs around the Link command, where it wouldn't change the nickname or wouldn't remove the nickname correctly

Also improved the code a little by doing some cleanup

KuryKat committed:
[CLEAN] Code improvement and cleanup

KuryKat committed:
[BUG] ServerStatus Command not replying back

Changed the way the Server Status Command works, instead of sending a message, it now replies to the command.

The Refresh button still replies with "Success" once it edits the original reply.

KuryKat committed:
[FEAT] Text Commands Deprecation @ v3.0.12

All Text commands are now DEPRECATED and should not be used.

Use Slash Commands Instead.

KuryKat commented:
Ready for review and merge, if any new bug shows up, let me know and I'll fix it right away!

hypherionmc commented on src/main/java/me/hypherionmc/sdlinklib/config/configobjects/BotSettingsConfig.java:
```diff
@@ -61,6 +61,10 @@ public class BotSettingsConfig {
 
+    @Path("botStatusStreamingURL")
+    @SpecComment("The URL that will be used when the \"botStatusType\" is set to \"STREAMING\", required to display as \"streaming\".")
+    public String botStatusStreamingURL = "https://twitch.tv/KuryKat";
```
Please don't use your own Twitch URL, before we get accused of advertising again. Is there any generic one we can use?

KuryKat replied:
Understood, I'm sorry, I was just trying to use some placeholder hehe 😅

KuryKat replied:
Maybe I can use the twitch URL: https://www.twitch.tv/twitch

hypherionmc replied:
Yeah, that would be fine

KuryKat replied:
Fixed in commit [eb413e39110b40f1e3f73a2804a95446390018e7](https://github.com/hypherionmc/sdlink-lib/pull/5/commits/eb413e39110b40f1e3f73a2804a95446390018e7)

hypherionmc commented on src/main/java/me/hypherionmc/sdlinklib/discord/commands/OfflineWhitelist.java:
```diff
@@ -169,16 +168,14 @@ protected void execute(CommandEvent event) {
+                            String nickName = event.getMember().getEffectiveName();
+                            if (pattern.matcher(nickName).find()) {
+                                try {
+                                    event.getMember().modifyNickname(null).queue();
```
Please don't clear the nickname. The user could've changed their nickname themselves, after account linking and clearing it would result in some unhappy people

KuryKat replied:
The nickname is only cleared if it matches the Pattern, which would be only if they have the [MC: Username], before this change I made, the nickname was being changed always to the same nickname, which means: it wasn't doing anything to the nickname after the unlink (the method "matches" only work if it matches the whole text, which was never the case, cause even the RegEx was incorrect with the nickname pattern.

hypherionmc commented on src/main/java/me/hypherionmc/sdlinklib/discord/commands/UnLinkCommand.java:
```diff
@@ -63,16 +63,14 @@ protected void execute(CommandEvent event) {
+            String nickName = event.getMember().getEffectiveName();
+            if (pattern.matcher(nickName).find()) {
+                try {
+                    event.getMember().modifyNickname(null).queue();
```
Same as before, don't clear the nickname

KuryKat replied:
Same as before

hypherionmc commented on src/main/java/me/hypherionmc/sdlinklib/discord/commands/WhitelistCommand.java:
```diff
@@ -173,16 +172,14 @@ protected void execute(CommandEvent event) {
+                            String nickName = event.getMember().getEffectiveName();
+                            if (pattern.matcher(nickName).find()) {
+                                try {
+                                    event.getMember().modifyNickname(null).queue();
```
Same as before, don't clear the nickname

KuryKat replied:
Same as before

hypherionmc commented on src/main/java/me/hypherionmc/sdlinklib/discord/slashcommands/LinkSlashCommand.java:
```diff
@@ -101,23 +100,21 @@ protected void execute(SlashCommandEvent event) {
+                String nickName = event.getMember().getEffectiveName();
+                if (pattern.matcher(nickName).find()) {
+                    try {
+                        event.getMember().modifyNickname(null).queue();
```
Same as before, don't clear the nickname

KuryKat replied:
Same as before

hypherionmc commented on src/main/java/me/hypherionmc/sdlinklib/discord/slashcommands/OfflineWhitelistSlashCommand.java:
```diff
@@ -205,16 +204,14 @@ protected void execute(SlashCommandEvent event) {
+                        String nickName = event.getMember().getEffectiveName();
+                        if (pattern.matcher(nickName).find()) {
+                            try {
+                                event.getMember().modifyNickname(null).queue();
```
Same as before, don't clear the nickname

KuryKat replied:
Same as before

hypherionmc commented on src/main/java/me/hypherionmc/sdlinklib/discord/slashcommands/WhitelistSlashCommand.java:
```diff
@@ -214,16 +213,14 @@ protected void execute(SlashCommandEvent event) {
+                        String nickName = event.getMember().getEffectiveName();
+                        if (pattern.matcher(nickName).find()) {
+                            try {
+                                event.getMember().modifyNickname(null).queue();
```
Same as before, don't clear the nickname

KuryKat replied:
Same as before

hypherionmc commented on src/main/java/me/hypherionmc/sdlinklib/discord/commands/ServerStatusCommand.java:
```diff
@@ -43,7 +44,8 @@ public ServerStatusCommand(BotController controller) {
     protected void execute(CommandEvent event) {
-        ServerStatusSlashCommand.runStatusCommand(minecraftHelper, event.getChannel(), null);
+        Button refreshButton = Button.danger("refreshbtn", "Refresh");
+        event.getChannel().sendMessageEmbeds(ServerStatusSlashCommand.runStatusCommand(minecraftHelper)).addActionRow(refreshButton).queue();
```
Won't this keep sending new embeds instead of updating the old embed?

KuryKat committed:
[PATCH] Change Placeholder URL

As requested in the Code Review, changing the default URL for the "botStatusStreamingURL" config.
